dexmaker-mockito/src/main/java/com/android/dx/mockito/InvocationHandlerAdapter.java

```diff
@@ -17,39 +17,41 @@
 package com.android.dx.mockito;
 
 import com.android.dx.stock.ProxyBuilder;
-import java.lang.reflect.InvocationHandler;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
-
+import org.mockito.internal.debugging.LocationImpl;
 import org.mockito.internal.invocation.InvocationImpl;
 import org.mockito.internal.invocation.MockitoMethod;
 import org.mockito.internal.invocation.realmethod.RealMethod;
 import org.mockito.internal.progress.SequenceNumber;
-import org.mockito.internal.util.ObjectMethodsGuru;
 import org.mockito.invocation.MockHandler;
 
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
 /**
  * Handles proxy method invocations to dexmaker's InvocationHandler by calling
  * a MockitoInvocationHandler.
  */
 final class InvocationHandlerAdapter implements InvocationHandler {
     private MockHandler handler;
-    private final ObjectMethodsGuru objectMethodsGuru = new ObjectMethodsGuru();
 
     public InvocationHandlerAdapter(MockHandler handler) {
         this.handler = handler;
     }
 
+    @Override
     public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
-        if (objectMethodsGuru.isEqualsMethod(method)) {
+        // args can be null if the method invoked has no arguments, but Mockito expects a non-null array
+        args = args != null ? args : new Object[0];
```

Comment:
The `InvocationImpl` class expects a non-null array of args, which is why i have to do this.